### PR TITLE
[1.14.x] ObfuscationReflectionHelper 

### PR DESCRIPTION
**Pull request by Cadiboo**
Added documentation for every method, removed unused fields, made the `findField` method public and added added another `findField` method that uses a field index instead of a name.

### CODE REVIEW

**Comment by williewillus** [comment]
The variants that take indices need to die in a fire. SRG names are all these should take.

**Comment by Cadiboo** [comment]
> The variants that take indices need to die in a fire. SRG names are all these should take.

I very much agree. Should I remove them or should that be part of another PR?

File: src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java [hunk 1]
```diff
@@ -211,6 +302,34 @@ private static Field findField(Class<?> clazz, String name)
...
+     * @param clazz      The class to find the field on.
+     * @param fieldIndex The index of the field on the class
+     * @param <T>        The type.
+     * @return The constructor with the specified parameters in the given class.
+     * @throws NullPointerException       If {@code clazz} is null.
+     * @throws UnableToFindFieldException If the field could not be found.
+     * @deprecated Use {@link #findField(Class, String)} because field indices change a lot more often than field names do.
+     */
+    @Deprecated
+    @Nonnull
```
Comment (LexManos) [inline src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java:323]: Don't ADD a deprecated entry...

Comment (Cadiboo) [reply src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java:323]: I added it for completeness, but it shouldn't be used. Should all the methods that use indices be removed?

Comment (LexManos) [reply src/main/java/net/minecraftforge/fml/common/ObfuscationReflectionHelper.java:323]: They can't be removed, as that breaks binary compatibility. But it needs a @Deprecated //Remove in 1.15 marker.

**Pull request merged**